### PR TITLE
API Restructure

### PR DESCRIPTION
**Pull request by n8maninger**
- Adds a bit more structure to wallets and wallet addresses with the goal of making usage between implementations more consistent.
  - Wallets now have a unique ID, friendly name, and description for easier updating.
  - Addresses (optionally) support storing their spend policy
- Splits the SC and SF utxo endpoints
- Adds pagination to the SC and SF utxo endpoints
- Returns 404 errors on all the wallet endpoints if the wallet does not exist

### CODE REVIEW

File: wallet/seed.go [hunk 1]
```diff
@@ -70,13 +70,19 @@ func (sav *SeedAddressVault) OwnsAddress(addr types.Address) bool {
...
-func (sav *SeedAddressVault) NewAddress(desc string) (types.Address, json.RawMessage) {
+func (sav *SeedAddressVault) NewAddress(desc string) Address {
 	sav.mu.Lock()
 	defer sav.mu.Unlock()
 	index := uint64(len(sav.addrs)) - sav.lookahead + 1
 	sav.gen(index + sav.lookahead)
-	addr := types.StandardAddress(sav.seed.PublicKey(index))
-	return addr, json.RawMessage(fmt.Sprintf(`{"desc":"%s","keyIndex":%d}`, desc, index))
+	policy := types.PolicyPublicKey(sav.seed.PublicKey(index))
+	addr := policy.Address()
```
Comment (lukechampine) [inline wallet/seed.go:79]: this reminds me that `SeedAddressVault` only generates v2-compatible addresses. I don't think we're actually using it anywhere, but it's something to keep in mind.

**Comment by lukechampine** [comment]
Good stuff. What's the motivation for names->IDs, especially `int64`? Is it a sqlite thing?

**Comment by n8maninger** [comment]
> Good stuff. What's the motivation for names->IDs, especially `int64`? Is it a sqlite thing?

name -> ID makes it possible to later change the name and keep the other wallet data consistent. The UI currently generates a UUID as the name and adds an additional friendlyName to metadata. This enforces that flow for all consumers.

I don’t really have a preference on the type. Int64 is easy and supported by every db engine I’m aware of. We could also use UUID or frand.Entry128().String. As long as it’s unique and set by `walletd`

**Comment by lukechampine** [comment]
Ok. Let's at least define a distinct `WalletID` type for it.

**Review by lukechampine — APPROVED** [review]

**Pull request merged**